cli.js

```diff
@@ -72,101 +92,81 @@ if (platform !== 'javascript') {
   });
 } else {
   if (compilerFlags.help === true) {
-    console.log('Sample usage: --compilation_level (-O) VAL --externs VAL --js VAL --js_output_file VAL --warning_level (-W) [QUIET | DEFAULT | VERBOSE]');
-    console.log('See https://github.com/google/closure-compiler/wiki/Flags-and-Options for the full list of flags');
+    process.stdout.write('Sample usage: --compilation_level (-O) VAL --externs VAL --js VAL --js_output_file VAL --warning_level (-W) [QUIET | DEFAULT | VERBOSE]\n');
+    process.stdout.write('See https://github.com/google/closure-compiler/wiki/Flags-and-Options for the full list of flags`);\n');
     process.exit(0);
   }
 
   if (compilerFlags.version === true) {
     const {version} = require('./package.json');
-    console.log(`Version: ${version}`);
+    process.stdout.write(`Version: ${version}\n`);
     process.exit(0);
   }
 
-  let inputFilePromises = [];
   let waitOnStdIn = true;
   if (compilerFlags.js) {
     waitOnStdIn = false;
-    if (!Array.isArray(compilerFlags.js)) {
-      compilerFlags.js = [compilerFlags.js];
-    }
-    inputFilePromises = compilerFlags.js.map(path =>
-        new Promise((resolve, reject) =>
-            fs.readFile(path, 'utf8', (err, src) => err ? reject(err) : resolve({src, path})))
-        .catch(e => {
-          process.exitCode = 1;
-          console.error(e);
-        }));
-
-    delete compilerFlags.js;
-  }
-  let externFilePromises = [];
-  if (compilerFlags.externs) {
-    if (!Array.isArray(compilerFlags.externs)) {
-      compilerFlags.externs = [compilerFlags.externs];
-    }
-    externFilePromises = compilerFlags.externs.map(path =>
-        new Promise((resolve, reject) =>
-            fs.readFile(path, 'utf8', (err, src) => err ? reject(err) : resolve({src, path})))
-        .catch(e => {
-          process.exitCode = 1;
-          console.error(e);
-        }));
-
-    delete compilerFlags.externs;
   }
 
-  Promise.all([...inputFilePromises, ...externFilePromises])
-    .then(files => {
-      const inputFiles = files.slice(0, inputFilePromises.length);
-      const externs = files.slice(inputFilePromises.length);
-      if (externs.length > 0) {
-        compilerFlags.externs = externs;
-      } else {
-        delete compilerFlags.externs;
+  const startCompile = !waitOnStdIn ? Promise.resolve([]) : new Promise(resolve => {
```

Comment:
Could you call this `getFilesFromStdin` or something like that instead, and possibly add a comment?